src/functions/GaussFunc.cpp

```diff
@@ -298,8 +316,23 @@ std::ostream& GaussFunc<D>::print(std::ostream &o) const {
  */
 template<>
 double GaussFunc<3>::calcCoulombEnergy(GaussFunc<3> &gf) {
-    double p = this->getExp();
-    double q = gf.getExp();
+
+    auto p_tmp = this->getExp();
+    auto is_all_same = std::all_of(p_tmp.begin(), p_tmp.end(),
```

Comment:
Given that this `all_of` pattern repeates, I suggest you factor it out into its own little function in a `detail` namespace.